### PR TITLE
[JENKINS-40808] Using docker-fixtures repository

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jglick/docker-fixtures/pull/1.

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java [hunk 1]
```diff
@@ -1,36 +0,0 @@
...
-import java.nio.charset.StandardCharsets;
-
-import org.apache.commons.io.FileUtils;
-
-/**
- * A DockerContainer whose <code>Dockerfile</code> is not static but requires some knowledge of the environment it is going to run in.
- *
- * Before the image is created {@link #process(String)} will be called which can be used to replace place holders with given values or manipulate the docker file in anyother way.
- */
-public class DynamicDockerContainer extends DockerContainer {
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java]: I wanted to get rid of this in favor of using build arguments. Would be great doing that before other projects start to depend on this behavior.

**Comment by olivergondza** [comment]
I am -0.5 on sharing the actual Dockerfiles. I am not confident we will be able to preserve needed compatibility in case we need to change some of the fixtures (update/rewrite/adding features).

File: src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java]: From what I can tell, @jtnord introduced this without any actual use cases, in #84. I have no opinion about it, and whether it should stay or be deleted seems orthogonal to this refactoring.

**Comment by jglick** [comment]
> I am -0.5 on sharing the actual Dockerfiles.

I am actually reusing `JavaContainer` (and thus by extension `SshdContainer`) from https://github.com/jenkinsci/mercurial-plugin/issues/93. Yes I could copy all that implementation if I had to, but I would really rather not.

And any functional tests which require a container for running an agent—typically because the plugin runs some special external commands as a build step, rather than making a socket connection to some kind of server, so in practice a large portion of plugins out there—will also need `JavaContainer`.

> I am not confident we will be able to preserve needed compatibility

Is it really a problem to deprecate a container and introduce a replacement? For that matter, this is just a test utility, so backward-incompatible changes are not that big a deal—you add some release notes, and if and when a plugin developer decides to upgrade the test dependency, either their tests continue to pass or they get a failure and figure out what they need to adapt to.

What is more irritating is that derived `Dockerfile`s currently need to hardcode the tag as produced by `Docker.getDockerFileHash`. Probably `build` could retag the newly produced image as `latest`, or simply delete `getDockerFileHash` and stop trying to use clever tags.

**Comment by jglick** [comment]
BTW these comments would probably be better placed in https://github.com/jglick/docker-fixtures/pull/1.

File: src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java (hunk 1 above)
Comment (jtnord) [inline src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java]: I found another way to accomplish what I required,  but given I had written it left it incase it was any use in the future.

**Comment by jglick** [comment]
@olivergondza ping?

**Comment by scoheb** [comment]
But do you not need https://issues.jenkins-ci.org/browse/HOSTING-254 first?

**Comment by batmat** [comment]
I suppose this can be merged and move forward now https://github.com/jglick/docker-fixtures has been forked as https://github.com/jenkinsci/docker-fixtures right?

**Comment by olivergondza** [comment]
@batmat, there do not seem to be a release, though.

**Comment by batmat** [comment]
Right @olivergondza but I was looking at the current PR and saw a dependency against 1.0-SNAPSHOT so that should still work. Though granted, we probably don't want ATH release be blocked because of a non-RELEASE dependency.

**Comment by jglick** [comment]
Yes I need to cut a release and integrate—on my to-do list.

**Comment by jglick** [comment]
Test failures look unrelated, so merging.

**Pull request merged**